plugins/inputs/tencentcloudcm/tencentcloudcm.go

```diff
@@ -0,0 +1,371 @@
+
+	DiscoveryInterval config.Duration `toml:"discovery_interval"`
+
+	Accounts []*Account `toml:"accounts"`
+
+	client cmClient
+
+	Log telegraf.Logger `toml:"-"`
+
+	windowStart  time.Time
+	windowEnd    time.Time
+	discoverTool *discoverTool
+}
+
+// Account defines a Tencent Cloud account
+type Account struct {
+	Name       string       `toml:"name"`
+	SecretID   string       `toml:"secret_id"`
+	SecretKey  string       `toml:"secret_key"`
+	Namespaces []*Namespace `toml:"namespaces"`
+
+	crs *common.Credential
+}
+
+// Namespace defines a Tencent Cloud CM namespace
+type Namespace struct {
+	Name    string    `toml:"namespace"`
+	Metrics []string  `toml:"metrics"`
+	Regions []*Region `toml:"regions"`
+}
+
+// Region defines a Tencent Cloud region
+type Region struct {
+	RegionName string              `toml:"region"`
+	Instances  []*monitor.Instance `toml:"instances"`
+}
+
+type metricObject struct {
+	Metric    string
+	Region    string
+	Namespace string
+	Account   *Account
+
+	isDiscovered bool
+
+	MonitorInstances []*monitor.Instance
+}
+
+type cmClient interface {
+	GetMetricObjects(t TencentCloudCM) []metricObject
+	NewClient(region string, crs *common.Credential, t TencentCloudCM) (monitor.Client, error)
+	NewGetMonitorDataRequest(namespace, metric string, instances []*monitor.Instance, t TencentCloudCM) *monitor.GetMonitorDataRequest
+	GatherMetrics(client monitor.Client, request *monitor.GetMonitorDataRequest, t TencentCloudCM) (*monitor.GetMonitorDataResponse, error)
+}
+
+// SampleConfig implements telegraf.Input interface
+func (t *TencentCloudCM) SampleConfig() string {
+	return `
+	## Endpoint to make request against, the correct endpoint is automatically
+	## determined and this option should only be set if you wish to override the
+	## default.
+	##   ex: endpoint = "tencentcloudapi.com"
+	# endpoint = ""
+  
+	# The default period for Tencent Cloud Cloud Monitor metrics is 1 minute (60s). However not all
+	# metrics are made available to the 1 minute period. Some are collected at
+	# 5 minute, 60 minute, or larger intervals.
+	# See: https://intl.cloud.tencent.com/document/product/248/33882
+	# Note that if a period is configured that is smaller than the default for a
+	# particular metric, that metric will not be returned by the Tencent Cloud API
+	# and will not be collected by Telegraf.
+	#
+	# Requested Tencent Cloud Cloud Monitor aggregation Period (required - must be a multiple of 60s)
+	# period = "5m"
+  
+	# Collection Delay (must account for metrics availability via Tencent Cloud API)
+	# delay = "0m"
+  
+	## Maximum requests per second. Note that the global default Tencent Cloud API rate limit is
+	## 20 calls/second (1,200 calls/minute), so if you define multiple namespaces, these should add up to a
+	## maximum of 20.
+	## See https://intl.cloud.tencent.com/document/product/248/33881
+	# ratelimit = 20
+  
+	# Timeout for http requests made by the Tencent Cloud client.
+	timeout = "5s"
+  
```

Comment:
I think this is a sensible default...
```suggestion
	## Timeout for http requests made by the Tencent Cloud client.
	# timeout = "5s"
  
```

Reply:
fixed in commit 880ecd1